src/gui/mzroll/database.cpp

```diff
@@ -850,12 +853,20 @@ int Database::loadCompoundCSVFile(string filename){
         if (id.empty()&& !name.empty()) id=name;
         if (id.empty() && name.empty()) id="cmpd:" + integer2string(loadCount);
 
-        if ( mz > 0 || ! formula.empty() || precursormz > 0) {
+        if ( mz > 0 || ! formula.empty() || precursormz > 0 || mass > 0.0f) {
             Compound* compound = new Compound(id,name,formula,charge);
 
             compound->expectedRt = rt;
 
-            if (mz == 0) mz = MassCalculator::computeMass(formula,charge);
+            if (mass != 0.0f)
+                compound->neutralMass = mass;
+
+            if (mz == 0 && !formula.empty()) {
+                mz = MassCalculator::computeMass(formula,charge);
+            } else if (mass != 0.0f) {
+                mz = MassCalculator::adjustMass(mass, 1);
```

Comment:
Why not send `charge` instead of 1 here?

Reply:
Will change.